### DISCUSSION TITLE
What happens if multiple commits should be applied?

### DISCUSSION THREAD

**Discussion by Thaval** [fluxcd/flux2#2706] [Q&A] [answered]
Hi, 

I wonder how flux works through the following workflow:
Image I commit and push a change to git and flux will, after the GitRepository's timeout of 1 minute, apply the changes to K8s.
What happens if I commit and push while the previous change is applied or what happens when the reconciliation takes longer than the timeout of GitRepository and Kustomization? Will it be applied afterwards or will the previous reconciliation be cancelled in order to reconcile the new change?

**Comment by stefanprodan** [comment, marked as answer]
All Flux controllers have an internal queue that prevents parallel reconciliations of the same object ID. This means previous reconciliations are not canceled, once a reconciliation finishes, the latest revision comes next, so if you push 10 times in a one minute interval, assuming the reconciliation takes 2 minutes, the first commit will be applied then the last one.

**Comment by Thaval** [comment]
Thanks for the quick response!
